### ISSUE TITLE
Handle failed XCM transfers when off-ramping from AssetHub

### ISSUE THREAD

**Issue by gianfra-t** [pendulum-chain/vortex#346]
### Issue description
A user attempted an off-ramp starting from AssetHub (XCM transaction [here](https://assethub-polkadot.subscan.io/extrinsic/7862371-2)) yet found that the application was stuck after signing.

This is most likely because we are not handling the case when the [transaction](https://github.com/pendulum-chain/vortex/blob/polygon-prototype-staging/src/services/phases/polkadot/assethub.ts#L56) is successful, yet the XCM execution itself is not. In this case, the application will wait indefinitely for the funds to arrive at Pendulum.

Also, this failed XCM should potentially have been prevented in the first place. The current theory is that it failed initially due to Existential Deposit issues, since the USDC amount attempted to be transferred was very close to the full amount the user had. A second [xcm](https://assethub-polkadot.subscan.io/extrinsic/7884348-2) with a lower amount worked well.

### Potential fixes
We could query the status of the XCM [here](https://github.com/pendulum-chain/vortex/blob/polygon-prototype-staging/src/services/phases/polkadot/assethub.ts#L61C11-L61C24), by listening to the events of that block and checking if the XCM failed.

Aside from that, we need to verify if the existential deposit was the root cause of the first failed off-ramp, and if so, consider it [here](https://github.com/pendulum-chain/vortex/blob/polygon-prototype-staging/src/hooks/useInputTokenBalance.ts#L30).

**Comment by gianfra-t** [comment]
@pendulum-chain/product although there is no risk of loosing funds if this happens again, we should consider fixing this to make Assethub off-ramps more robust. It is very likely users will try to off-ramp all it's USDC and this happens again.

**Comment by prayagd** [comment]
@gianfra-t is there a ED for USDC too on Assethub? if that's the case then a quick solution would be to show the balance minus ED on the vortex UI

**Comment by gianfra-t** [comment]
Yup, that's a good simple solution. I think there is an ED for any token. What remains to be seen is if it works if all balance is transferred, we may have to use the `transfer_all` in this case to allow for people to do that.
